packages/grid_client/tests/zos/deployments.test.ts

```diff
@@ -0,0 +1,827 @@
+    });
+
+    it("should handle empty Uint8Array", () => {
+      const deployment = createDeployment();
+      const result = deployment.to_hex(new Uint8Array(0));
+      expect(result).toBe("");
+    });
+
+    it("should handle invalid hex string with non-hex characters", () => {
+      const deployment = createDeployment();
+      const invalidHex = "48656c6c6g";
+      const result = deployment.from_hex(invalidHex);
+
+      expect(result).toEqual(new Uint8Array([72, 101, 108, 108, 6]));
+    });
+
+    it("should handle invalid hex string with odd length", () => {
+      const deployment = createDeployment();
+      const invalidHex = "48656c6c6";
+      const result = deployment.from_hex(invalidHex);
+
+      expect(result).toEqual(new Uint8Array([72, 101, 108, 108]));
+    });
+  });
+
+  describe("validation", () => {
+    it("should validate with valid data", async () => {
+      const deployment = createDeployment();
+
+      const errors = await validate(deployment);
+      expect(errors.length).toBe(0);
+    });
+
+    it("should fail validation with invalid version", async () => {
+      const deployment = createDeployment({ version: -1 });
+
+      const errors = await validate(deployment);
+      expect(errors.length).toBeGreaterThan(0);
+      expect(errors[0].constraints?.min).toBeDefined();
+    });
+
+    it("should fail validation with invalid twin_id", async () => {
+      const deployment = createDeployment({ twin_id: 0 });
+
+      const errors = await validate(deployment);
+      expect(errors.length).toBeGreaterThan(0);
+      expect(errors[0].constraints?.min).toBeDefined();
+    });
+
+    it("should fail validation with null twin_id", async () => {
+      const deployment = createDeployment({ twin_id: null as any });
+
+      const errors = await validate(deployment);
+      expect(errors.length).toBeGreaterThan(0);
+    });
+
+    it("should fail validation with undefined twin_id", async () => {
+      const deployment = createDeployment({ twin_id: undefined as any });
+
+      const errors = await validate(deployment);
+      expect(errors.length).toBeGreaterThan(0);
+    });
+  });
+});
+
+describe("Deployment signing", () => {
+  let deployment: Deployment;
+
+  const verifySignature = async (
+    message: string,
+    signature: string,
+    mnemonic: string,
+    keypairType: KeypairType,
+  ): Promise<boolean> => {
+    try {
+      await waitReady();
+      const keyring = new Keyring({ type: keypairType });
+      const keypair = keyring.addFromUri(mnemonic);
+
+      const signatureBytes = new Uint8Array(signature.length / 2);
+      for (let i = 0; i < signature.length / 2; i++) {
+        signatureBytes[i] = parseInt(signature.slice(2 * i, 2 * i + 2), 16);
+      }
+
+      const messageBytes = new Uint8Array(message.length / 2);
+      for (let i = 0; i < message.length / 2; i++) {
+        messageBytes[i] = parseInt(message.slice(2 * i, 2 * i + 2), 16);
+      }
```

Comment:
this should probably be extracted in a util file as it may be reused 
also 
```suggestion
        const signatureBytes = new Uint8Array(signature.length / 2);
for (let i = 0; i < signature.length; i += 2) {
  signatureBytes[i / 2] = parseInt(signature.slice(i, i + 2), 16);
}

const messageBytes = new Uint8Array(message.length / 2);
for (let i = 0; i < message.length; i += 2) {
  messageBytes[i / 2] = parseInt(message.slice(i, i + 2), 16);
}
```